independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/AsyncInvokerImpl.java

```diff
@@ -280,12 +281,22 @@ RestClientRequestContext performRequestInternal(String httpMethodName, Entity<?>
     public <T> CompletableFuture<T> mapResponse(CompletableFuture<Response> res, Class<?> responseType) {
         if (responseType.equals(Response.class)) {
             return (CompletableFuture<T>) res;
+        } else if (responseType.equals(RestResponse.class)) {
+            return res.thenApply(new Function<>() {
+                @Override
+                public T apply(Response response) {
+                    return (T) RestResponse.ResponseBuilder.create(response.getStatusInfo(), response.getEntity())
+                            .replaceAll(response.getHeaders()).build();
```

Comment:
if you do a rework, could you rename `replaceAll` to `replaceAllHeaders` or sth else that would be descriptive?

Reply:
The name was kept the same as `javax.ws.rs.core.Response`.  Stef tried to keep things the same as much as possible.

Reply:
That is a really weird choice (by the JAX-RS people)...

Reply:
I wish it were the only one :)